### PR TITLE
[E2E] Rewrite HostGroupConfiguration.feature

### PR DESCRIPTION
**Pull request by s-elhassani**
## Description

Rewrite the behat test for host groups configuration with Cypress (add Gherkin file and Implementation files)

### CODE REVIEW

File: centreon/tests/e2e/features/HostGroups/01-host-group-configuration.feature [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    When the user changes some properties of the configured host group "name, alias, comments"
+    Then the properties are updated
+
+  Scenario: Duplicate one existing host group
+    When the user duplicates the configured host group
+    Then a new host group is created with identical properties
+
+  Scenario: Delete one existing host group
+    When the user deletes the configured host group
+    Then the configured host group is not visible anymore on host group page
```
Comment (mushroomempires) [inline centreon/tests/e2e/features/HostGroups/01-host-group-configuration.feature:20]:
```suggestion
    Then the configured host group is not visible anymore on the host group page
```

File: centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        cy.getIframeBody().find('div.md-checkbox.md-checkbox-inline').eq(2).click();
+        cy.getIframeBody().find('select').eq(0)
+            .invoke('attr', 'onchange', "javascript: { setO(this.form.elements['o1'].value); this.form.submit(); }");
+        cy.getIframeBody().find('select').eq(0).select('Delete');
+        cy.wait('@getTimeZone');
+        cy.exportConfig();
+    }
+);
+
+Then('the configured host group is not visible anymore on host group page',
```
Comment (mushroomempires) [inline centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts:167]:
```suggestion
Then('the configured host group is not visible anymore on the host group page',
```

File: centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        cy.getIframeBody().find('textarea[name="hg_comment"]').clear().type('Host group updated for test!');
+        cy.getIframeBody().find('input.btc.bt_success[name^="submit"]').eq(0).click();
+        cy.wait('@getTimeZone');
+    }
+);
+
+Then('the properties are updated',
+    () => {
+        cy.exportConfig();
+        cy.getIframeBody().contains(hostGroup + '_new').should('exist');
```
Comment (kduret) [inline centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts:139]: where do you check that properties are updated ?

File: centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+            .invoke('attr', 'onchange', "javascript: { setO(this.form.elements['o1'].value); this.form.submit(); }");
+        cy.getIframeBody().find('select').eq(0).select('Duplicate');
+        cy.wait('@getTimeZone');
+        cy.exportConfig();
+    }
+);
+
+Then('a new host group is created with identical properties',
+    () => {
+        cy.getIframeBody().contains(hostGroup + '_1').should('exist');
```
Comment (kduret) [inline centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts:162]: where do you check that properties are identical ?

**Review by kduret — COMMENTED** [review]
You should check the coding style with `eslint` because it seems there are some issues

File: centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts (hunk 3 above)
Comment (s-elhassani) [reply centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts:154]: after update, the name of the host group will be hostGroup + '_new, so in the page should contains hostGroup + '_new

File: centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts (hunk 4 above)
Comment (s-elhassani) [reply centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts:169]: the new duplicate host group has the same name as the first but a '_1' is added

Comment (kduret) [reply centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts:169]: you are replacing the work `properties` by `name` :)
`properties` means name, alias, address, snmp community, linked templates...

Comment (s-elhassani) [reply centreon/tests/e2e/features/HostGroups/01-host-group-configuration/index.ts:169]: this test is based on what it has been done for behat test, they check just that they have the same name

**Review by mushroomempires — DISMISSED** [review]

File: centreon/tests/e2e/features/Hosts/commands.ts [hunk 5]
```diff
@@ -33,6 +33,47 @@ Cypress.Commands.add(
...
+    .then(($label) => {
+      const radioId = $label.attr('for');
+      cy.getIframeBody().find(`input[type="radio"][id="${radioId}"]`)
+        .should('be.checked');;
+    });
+});
+
+Cypress.Commands.add('updateHostGroupViaApi', (body: HostGroup, hostGroup_name: string) => {
+  let query =
+    `SELECT h.hg_id  from hostgroup as h WHERE h.hg_name='${hostGroup_name}'`;
```
Comment (kduret) [inline centreon/tests/e2e/features/Hosts/commands.ts:48]:
```suggestion
    `SELECT h.hg_id from hostgroup as h WHERE h.hg_name='${hostGroup_name}'`;
```

File: centreon/tests/e2e/features/Hosts/commands.ts [hunk 6]
```diff
@@ -33,6 +33,47 @@ Cypress.Commands.add(
   }
 );
 
+Cypress.Commands.add('isEnableOrDisableResourceOnHostGroupChecked', (label: string) => {
+  cy.getIframeBody().contains('label', label)
+    .should('exist')
+    .then(($label) => {
+      const radioId = $label.attr('for');
+      cy.getIframeBody().find(`input[type="radio"][id="${radioId}"]`)
+        .should('be.checked');;
```
Comment (kduret) [inline centreon/tests/e2e/features/Hosts/commands.ts:42]:
```suggestion
        .should('be.checked');
```

File: centreon/tests/e2e/features/Hosts/commands.ts [hunk 7]
```diff
@@ -33,6 +33,47 @@ Cypress.Commands.add(
   }
 );
 
+Cypress.Commands.add('isEnableOrDisableResourceOnHostGroupChecked', (label: string) => {
```
Comment (kduret) [inline centreon/tests/e2e/features/Hosts/commands.ts:36]:
```suggestion
Cypress.Commands.add('checkLegacyRadioButton', (label: string) => {
```

**Review by kduret — APPROVED** [review]

**Review by nfragnet — APPROVED** [review]

**Review by MedWassCh — APPROVED** [review]

**Pull request merged**